Yafc/Workspace/ProductionTable/ProductionTableView.cs

```diff
@@ -306,10 +306,10 @@ private class EntityColumn : ProductionTableDataColumn {
                 }
             }
 
-            private void ShowAccumulatorDropdown(ImGui gui, RecipeRow recipe, Entity accumulator) {
+            private void ShowAccumulatorDropdown(ImGui gui, RecipeRow recipe, Entity oldAccumulator) {
                 gui.ShowDropDown(imGui => {
                     imGui.BuildInlineObjectListAndButton<EntityAccumulator>(Database.allAccumulators, DataUtils.DefaultOrdering,
-                        accumulator => recipe.RecordUndo().ChangeVariant(accumulator, accumulator), "Select accumulator",
+                        newAccumulator => recipe.RecordUndo().ChangeVariant(oldAccumulator, newAccumulator), "Select accumulator",
```

Comment:
I would rename `oldAccumulator` to `currentAccumulator`.

> I don't understand how exactly this tangle of functions works,
but the fix works and the bug is gone.

Problem is that the lamba function uses/used the same parameter (name) as `ShowAccumulatorDropdown()`. So the value of `ShowAccumulatorDropdown()` would never reach `ChangeVariant()` as it gets 'shadowed' by the lambda function parameter.

TBH I could not reproduce the issue with your steps, but this change makes sense in my mind.

Reply:
If it's not reproduced with my steps, then I would suggest to change the accumulators a couple of times. On the second or third time, my YAFC broke and didn't change them anymore without the fix.

Reply:
Renamed `oldAccumulator` to `currentAccumulator` in https://github.com/have-fun-was-taken/yafc-ce/pull/116/commits/6c61e3aa7a193c903bff7f46cb694cebe94601fa

Reply:
> Problem is that the lamba function uses/used the same parameter (name)

That I understood. What I haven't understood before is that the lambda inside `ShowDropDown` is `GuiBuilder`. Thank you for the motivation to dig deeper :D